hyperjumptech/monika
# Refactor: Encapsulate Database Global Variable

# Monika Pull Request (PR)

## What feature/issue does this PR add
1. Prevent database global variable mutate directly.
2. Remove database error when running test. Resolves #1137.

## How did you implement / how did you fix it
- refactor: create an encapsulating function to access and update database variable.
- refactor: replace reference to the database variable to the encapsulating function.
- refactor: restrict the visibility of the db variable.
- test: add open and close database in the test.
- test: fix flacky mongo db test due to using the same port.

## How to test
1. Run Monika as usual
2. See the logs. Run `npm start -- --logs`

## Screenshot
### Manual Test
![image](https://github.com/hyperjumptech/monika/assets/15191978/147b5268-19f8-4df5-a061-22dd7918572c)

### Database Error
#### Before
![image](https://github.com/hyperjumptech/monika/assets/15191978/0da4d17c-5007-41ce-82fe-8f55158fde29)

#### After
![image](https://github.com/hyperjumptech/monika/assets/15191978/f79b9b05-2b43-4147-9b06-67a099405a26)
